src/main/webapp/homepage_script.js

```diff
@@ -113,21 +81,37 @@ if (JSON.stringify(neighborhood) != JSON.stringify([null, null])) {
     }
 }
 
-/* Function that display the help out confirmation overlay */
+/* Function that display the help out overlay */
 function helpOut(element) {
     const task = element.closest(".task");
-    const overlay = task.getElementsByClassName("confirm-overlay");
+    const overlay = task.getElementsByClassName("help-overlay");
     overlay[0].style.display = "block";
 }
 
-/* Function that hides the task after user confirms they want to help out */
-function removeTask(element) {
-    element.closest(".task").style.display = "none";
+/* Function sends a fetch request to the edit task servlet when the user
+offers to help out, edits the task's status and helper properties, and
+then reloads the task list */
+function confirmHelp(element) {
+    const task = element.closest(".task");
+    const url = "tasks/edit?task-id=" + task.dataset.key + "&action=helpout";
```

Comment:
Also, it's more typical for query parameters in POST requests to appear in the body of the request, rather than the URL. Either way is fine, just letting you know.

To put the data in the body, you'd do something like:
```
const data = {"task-id": task.dataset.key, "action": "helpout"}
const request = new Request(url, {
    method: "POST",
    body: JSON.stringify(data),
    headers: {
        'Content-Type': 'application/json'
    }});
```

Some more examples at https://attacomsian.com/blog/using-javascript-fetch-api-to-get-and-post-data#using-fetch-to-post-data

I don't think you need to change this, but I wanted to bring it up. When you start having lots of parameters, putting them all in the URL starts to get more clunky.

Reply:
This works better, I will make both of these changes :)
Thanks!

Reply:
For some reason when I try using this method the fetch fails and throws the following error: `POST https://8080-dfe2475d-f262-4cff-8707-9112c4571804.us-west1.cloudshell.dev/tasks/edit 500 (Internal Server Error)` - strange. I'll keep it as before for now but I'm not really sure why this wasn't working.

Reply:
I'd check your server logs to see what the Java code is reporting. Usually a 500 error means something went wrong server-side.

Reply:
It seems the parameters aren't being passed through in the request. I wrote a debugging if-statement that prints out a message if the request doesn't find parameters for "task-id" and it printed out the message. The 500 error was happening because of a null pointer exception, from the missing parameter. The network tab also doesn't show the parameters in the request =/

Reply:
Hmm, it's probably not worth dwelling on then. We could debug it together at some point if you like, but we can move on for now.

Reply:
Sounds good!